include/grpcpp/impl/codegen/rpc_service_method.h

```diff
@@ -58,18 +59,57 @@ class RpcServiceMethod : public RpcMethod {
   /// Takes ownership of the handler
   RpcServiceMethod(const char* name, RpcMethod::RpcType type,
                    MethodHandler* handler)
-      : RpcMethod(name, type), server_tag_(nullptr), handler_(handler) {}
+      : RpcMethod(name, type),
+        server_tag_(nullptr),
+        async_type_(AsyncType::UNSET),
+        handler_(handler) {}
+
+  enum class AsyncType {
```

Comment:
Out of curiosity, why enum class and not just enum? We aren't really using those elsewhere afaik and you're just using them as int's and not arbitrary constexprs

Reply:
I was asked to use enum class in a different review. Usage [here](https://github.com/grpc/grpc/blob/master/src/core/lib/channel/channel_trace.h#L78). That is the only reason.

I will flip the question; are there reasons to prefer normal enum over enum class?

Reply:
No objections in retrospect. We should increasingly move to enum class whenever possible.